.azure-pipelines/pipelines.yml

```diff
@@ -572,6 +572,42 @@ stages:
         AZP_SHA1: $(Build.SourceVersion)
         DOCKERHUB_USERNAME: $(DockerUsername)
         DOCKERHUB_PASSWORD: $(DockerPassword)
+        WINDOWS_BUILD_TYPE: "windows"
+        WINDOWS_IMAGE_BASE: "mcr.microsoft.com/windows/servercore"
+        WINDOWS_IMAGE_TAG: "ltsc2019"
+    - task: PublishBuildArtifacts@1
+      inputs:
+        pathtoPublish: "$(Build.StagingDirectory)/build_images"
+        artifactName: docker_windows
+      condition: always()
+  - job: dockerWindowsNanoServer2022
```

Comment:
can we use a build matrix for this instead of duplicating?

Reply:
There are multiple reasons we can't use buildx to do this:

1. buildx is not supported on Windows
2. We rely on RUN commands on the Dockerfile which prohibits us from running buildx on linux. See https://github.com/envoyproxy/envoy/issues/17656
3. Building with buildx on linux would push windows foreign layers on our dockerhub and this would violate the license of the base layer.

Also Windows server images on ci do not come with hyper-v enabled so have to use two different machines for the two different versions of the OS.

Reply:
Resolved, thanks for the pointer offline